docs/tokens.md

```diff
@@ -0,0 +1,327 @@
+- **On-chain part (gas costs):** for every RIF Rollup block, the validator must pay Rootstock gas to verify the SNARK,
+  plus additional ~0.4k gas per transaction to publish the state 𝛥. The on-chain part is a variable that depends on the
+  current gas price in the Rootstock network. However, this part is orders of magnitude cheaper than the cost of normal
+  RBTC/ERC20 transfers.
+
+### **How fees are paid**
+
+Transfers in RIF Rollup support "gasless meta-transactions": users pay transaction fees in the tokens being transferred.
+Thus, for example, if you want to transfer RIFPro stable-coin, there is no need for you to own RBTC or any other tokens.
+Just pay your fees in a fraction of RIFPro.
+
+Tokens listed on Coingecko with $100 of total volume in usd (our current measure of liquidity) can be used to pay for
+fees. For tokens without a liquid market volume, fees are paid in a different token.
+
+This threshold can be also configured in `etc/env/<network>/fee_ticker.toml`, where `network` is either _testnet_,
+_mainnet_, or _dev_ (representing localhost), under:
+
+```toml
+liquidity_volume=100
+```
+
+## **Transaction batching**
+
+### **_JS SDK side_**
+
+Transaction batching allows the user to submit several transactions at the same time, with the added benefit that the
+collected fee may be calculated for the whole batch instead of per transaction.
+
+This allows the user to include any transaction in the batch, whether it operates with a token that is eligible for fee
+payment or not, since the batch builder allows the user to specify a fee token type that is independent of any of the
+tokens used in the operations. The user includes several transactions in the batch, each of them with ‘0’ as fee, and
+then selects a valid fee token when building the Batch transaction using the SDK, then the server will calculate the
+total amount needed to be paid on that fee token.
+
+For example, for two Transfer operations.
+
+```javascript
+const batch = await sender
+  .batchBuilder()
+  .addTransfer({ to: receiver.address(), tokenWithoutLiquidity1, amount })
+  .addTransfer({ to: receiver.address(), tokenWithoutLiquidity2, amount })
+  .build(tokenForFees);
+```
+
+By specifying the fee token (_tokenForFees_) during the Batch build process, it is expected for all the transactions
+included to have the “fee amount” in 0, otherwise, the build process will fail. The build process will
+[automatically calculate](https://github.com/rsksmart/rif-rollup/blob/main/sdk/zksync.js/src/batch-builder.ts#L110) the
+cost for the whole batch, in the provided _tokenForFees_ token. For that it will first calculate the fee with the
+[following formula](https://github.com/rsksmart/rif-rollup/blob/main/core/bin/zksync_api/src/fee_ticker/mod.rs#L408):
+
+> $
+> feeTokenRiskFactor = 1
+> $
+>
+> $
+> feeTokenPriceWithRiskInUSD = tokenRiskFactor/TokenPriceInUSD
+> $
+>
+> $
+> totalZKPFeeInTokenUnits = costPerChunkInUSD _totalChunks_ tokenPriceWithRiskInUSD
```

Comment:
```suggestion
> totalZKPFeeInTokenUnits = costPerChunkInUSD * _totalChunks_ * tokenPriceWithRiskInUSD
```

Reply:
lol how did this happen? Perhaps `zk fmt` removed it. I'll check

Reply:
done